lib/action_controller/serialization.rb

```diff
@@ -43,6 +44,12 @@ def get_serializer(resource, options = {})
       end
     end
 
+    def _handle_errors(serializable_resource)
+      return unless serializable_resource.errors?
+      return unless (200..299).cover?(response.status)
+      response.status = 422
```

Comment:
This makes me a little nervous, as it's hard-coded an not really opt-out (except by overwriting the method), but if we're going to return errors, we should return a proper response code.

I'm not sure about this type of automatic error handling.  Rails traditional separates the 'render' of success from 'failure', and I think this is a good pattern.

- See [api_controller template](https://github.com/spastorino/rails/blob/51d5d6252ee093d3fa004abec79319ae8b4c42c4/railties/lib/rails/generators/rails/scaffold_controller/templates/api_controller.rb)

Alternative methods requiring exception handling as flow control (yuck)

- See [rescue_with_handler](https://github.com/rails/rails/blob/e670611e6002039231a24d547f9a6e053940fb16/actionpack/lib/action_controller/metal/rescue.rb]

Here's an example we (I) removed from AMS:

```ruby
class RescueFromTestController < ActionController::Base
  rescue_from Exception, with: :handle_error

  def render_using_raise_error_serializer
    profile = Profile.new({ name: 'Name 1', description: 'Description 1', comments: 'Comments 1' })
    render json: [profile], serializer: RaiseErrorSerializer
  end

  # Can take an exception object or a decorated exception, e.g.
  # class ExceptionModel
  #   include ActiveModel::Serialization
  #   def initialize(exception)
  #     # etc
  #   end
  # end
  def handle_error(exception)
    render json: { ExceptionModel.new(exception) }, status: :internal_server_error
  end
end
```

and then in ActionController::Serialization

```ruby

  def rescue_with_handler(exception)
    super(exception)
  end
 ```  

or as jsonapi-resource does
```
    # Note: Be sure to either call super(e) or handle JSONAPI::Exceptions::Error and raise unhandled exceptions
    def handle_exceptions(e)
      case e
      when JSONAPI::Exceptions::Error
        render_errors(e.errors)
      else # raise all other exceptions
        # :nocov:
        fail e
        # :nocov:
      end
    end
```